src/network/networkd-link.c

```diff
@@ -1325,7 +1315,11 @@ static int link_reconfigure_impl(Link *link, bool force) {
                         return r;
         }
 
-        link_free_carrier_maps(link);
+        /* The bound_to map depends on .network file, hence it needs to be freed.
+         * But, do not free the bound_by map, Moreover, it is not necessary to recreate the map here,
```

Comment:
`/map,/map./` Also it would be good to mention what the commit says here, ie that bound_by cannot be freed or the state won't propagate